src/ngraph/runtime/backend.hpp

```diff
@@ -142,6 +142,18 @@ class ngraph::runtime::Backend
     /// \returns a shared pointer to the op if found, else nullptr
     virtual std::shared_ptr<ngraph::Node> get_backend_op(const std::string& op_name, ...);
 
+    /// \brief Allows sending backend specific configuration. The map contains key, value pairs
+    ///     specific to a particluar backend. The definition of these key, value pairs is
+    ///     defined by each backend.
+    /// \param config The configuration map sent to the backend
+    /// \param error An error string describing any error encountered
+    /// \returns true if the configuration is supported, false otherwise. On false the error
+    ///     parameter value is valid.
+    virtual bool set_config(const std::map<std::string, std::string>& config, std::string& error);
+
+    static void set_backend_shared_library_search_directory(const std::string& path);
```

Comment:
Please provide comments. Also, shouldn't this be cumulative i.e., a list of paths as opposed to just one path. That would be more flexible (i.e., various backends could come from various directories - something we already run into during testing).